pootle/core/search/backends/elasticsearch.py

```diff
@@ -26,6 +28,26 @@
 logger = logging.getLogger(__name__)
 
 
+def filter_hits_by_distance(hits, source_text, max_distance=0.3):
+    """Returns ES `hits` filtered according to their Levenshtein distance
+    to the `source_text`.
+
+    Any hits with a normalized distance value (0..1) higher than
+    `max_distance` will be discarded. It's assumed that `hits` is already
+    sorted from higher to lower score.
```

Comment:
It's perfectly fine to assume that. Even if you multiply the score by a weight factor later, those relevance scores are still a) sorted, b) refer to irrelevant results.